falcon/hooks.py

```diff
@@ -57,28 +72,27 @@ def _before(responder_or_resource):
         if isinstance(responder_or_resource, six.class_types):
             resource = responder_or_resource
 
-            for method in COMBINED_METHODS:
-                responder_name = 'on_' + method.lower()
+            members = inspect.getmembers(resource, predicate=predicate_method_function)
 
-                try:
-                    responder = getattr(resource, responder_name)
-                except AttributeError:
-                    # resource does not implement this method
-                    pass
-                else:
-                    # Usually expect a method, but any callable will do
-                    if callable(responder):
-                        # This pattern is necessary to capture the current
-                        # value of responder in the do_before_all closure;
-                        # otherwise, they will capture the same responder
-                        # variable that is shared between iterations of the
-                        # for loop, above.
-                        def let(responder=responder):
-                            do_before_all = _wrap_with_before(responder, action, args, kwargs)
+            responders = []
 
-                            setattr(resource, responder_name, do_before_all)
+            for member in members:
+                for method in COMBINED_METHODS:
+                    if member[0].startswith('on_' + method.lower()):
```

Comment:
Probably ``'on_' + method.lower() + '_'`` to make it even more specific, if one wants to go this way.

Reply:
Ah, true.
What I trying to suggest though was that we should match ``on_post`` and ``on_post_suffixed``, but not ``on_postman``.